src/hw_stats.py

```diff
@@ -143,19 +241,92 @@ def get_hw_stats (fid, node_dut, cycle):
     hw_comp_access['tile_control'] = sum_num_cycle_ima
 
     total_energy = 0
+    total_adc_energy = 0
+    total_adc_access = 0
+    total_mvm_energy = 0
+    total_mvm_access = 0
+    total_mtvm_access = 0
+    total_mtvm_energy = 0
+    total_op_access = 0
+    total_op_energy = 0
     # Compute the total dynamic energy consumption
-    for key, value in hw_comp_access.items():
-        total_energy += value * hw_comp_energy[key]
+    if cfg.MVMU_ver == "Analog":
```

Comment:
Why is this `if, else` block required, if specific component energy (based on `mvmu_ver`) and their access are already resolved before ?